File: final project - social media issue posting app/springapp/src/main/java/com/application/Issue/Service/PollService.java
package com.application.Issue.Service;

import java.util.List;
import java.util.Map;

import com.application.Issue.Model.Poll;

public interface PollService {
    
    public Poll addPoll(Poll poll);
    public Poll updatePoll(Poll poll);
    public List<Poll> findAllPolls();
    public List<Poll> getPollsByUserId(String userId);
    public List<Poll> getPollsByIssueId(String issueId);
    public List<Poll> getPollsByIssueIdAndUserID(String issueId, String userId);
    public Map<String, Map<String, Double>> getOptionPercentagesByIssue();
}

File: final project - social media issue posting app/angularapp/src/app/services/poll.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Poll } from '../models/poll';

@Injectable({
  providedIn: 'root'
})
export class PollService {

  private baseUrl='http://localhost:8080/poll';
  
  constructor(private http: HttpClient) { }

  public addPoll(poll: Poll): Observable<Poll> {
    let url = `${this.baseUrl}`;
    return this.http.post<Poll>(url, poll);
  }

  //to get issues based on userId
  public getPollsByUserId(userId:string): Observable<Poll[]> {
    let url = `${this.baseUrl}/userId/${userId}`;
    return this.http.get<Poll[]>(url);
  }
  
  //to get issues based on issueId
  public getPollsByIssueId(issueId: string | undefined): Observable<Poll[]> {
    let url = `${this.baseUrl}/issueId/${issueId}`;
    return this.http.get<Poll[]>(url);
  }

}
